### PR TITLE
DOC: Adds Inigo's week 8&9 blogposts 

### PR DESCRIPTION
**Pull request by itellaetxe**
Also and adapts image paths to new dedicated folder (thanks @deka27 for creating it)

3 images from past posts had not been moved to the dedicated folder, so I moved them.

### CODE REVIEW

File: posts/2024/2024_07_26_Inigo_week_9.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    :author: Iñigo Tellaetxe
+    :tags: google
+    :category: gsoc
+
+What I did this week
+~~~~~~~~~~~~~~~~~~~~
+
+This week was a bit shorter than usual because Thursday was a holiday in the Basque Country, and today we had an outdoor activity with my lab mates (we went kayaking to the Urdaibai Biosphere Reserve). Nevertheless, it was full of advances and interesting scientific matters.
+
+As I mentioned in my last blog post, this week I worked on the implementation of the conditional VAE based on `this implementation <https://github.com/QingyuZhao/VAE-for-Regression/tree/master>` of this `paper <https://doi.org/10.1007/978-3-030-32245-8_91>`.
```
Comment (skoudoro) [inline posts/2024/2024_07_26_Inigo_week_9.rst:14]: link does not work. wrong formatting I suppose. see here: https://dipy.github.io/preview-html/dipy.org/pull/55/posts/2024/2024_07_26_Inigo_week_9.html

**Review by skoudoro — COMMENTED** [review]
Hi @itellaetxe,

Very interesting blog post. I learn new stuff so well explained. 
See below a small comment.

also, we sure to add more spacing to make it easier to read. I know you already made a effort but I think you can add more paragraph and let the text breathe.

see your blog post preview here: https://dipy.github.io/preview-html/dipy.org/pull/55/posts/2024/2024_07_26_Inigo_week_9.html.   or this one: https://dipy.github.io/preview-html/dipy.org/pull/55/posts/2024/2024_07_19_Inigo_week_8.html

Thank you in advance for updating that also

**Comment by skoudoro** [comment]
> 3 images from past posts had not been moved to the dedicated folder, so I moved them.

great catch! thanks !

File: posts/2024/2024_07_26_Inigo_week_9.rst (hunk 1 above)
Comment (itellaetxe) [reply posts/2024/2024_07_26_Inigo_week_9.rst:14]: Solved

**Comment by itellaetxe** [comment]
@skoudoro, I am getting this error while building locally:
image
I investigated it on the internet but I cannot find a reasonable solution.

On the other hand, the CI fails but I am not familiar with this error:
image

I think it has to do with how `make -C . html` works. When the `html` command is issued, `$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) _build/html` is executed. However, `$(ALLSPHINXOPTS)` is defined as `-d _build/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .`, but if $(PAPER)` is not defined, does the html command get executed?

I have barely used Makefiles, so I don't know if this a silly question 😅

File: posts/2024/2024_07_19_Inigo_week_8.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+~~~~~~~~~~~~~~~~~~~~
+
+This week I continued training the VAE model with the FiberCup dataset, this time for 120 epochs, and the results are promising. The model is able to reconstruct the input data with a decent level of detail.
+
+.. image:: /_static/images/gsoc/2024/inigo/vanilla_vae_120_epoch_results.png
+    :alt: Vanilla Variational AutoEncoder reconstruction results after 120 epochs of training.
+    :width: 600
+
+I also looked at the theoretical and technical implications of implementing the `beta-VAE architecture <https://openreview.net/forum?id=Sy2fzU9gl>`_ for my experiments, which could help in disentangling the latent space representation of the streamlines according to features learnt in an
+unsupervised manner.
```
Comment (jhlegarreta) [inline posts/2024/2024_07_19_Inigo_week_8.rst:19]: Unnecessary line break

File: posts/2024/2024_07_19_Inigo_week_8.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+Shortly, applying a weight (bigger than 1) to the KL loss component of the VAE loss function encourages the model to learn a version of the latent space where features that can be perceived in the data space are aligned with the latent space dimensions. This way, one can modulate the generative process according to the learnt 'perceivable' features, once they are identified and located in the latent space.
+
+However, increasing the beta weight compromises the reconstruction quality, which is what basically makes streamlines look reasonable. Finding a good beta weight is as 'simple' as running a hyperparameter search while constraining the parameter to be higher than one, and to try to prioritize
+the MSE (Mean Squared Error, reconstruction loss) in the search algorithm.
```
Comment (jhlegarreta) [inline posts/2024/2024_07_19_Inigo_week_8.rst:24]: Unnecessary line break

File: posts/2024/2024_07_19_Inigo_week_8.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+From the technical side implementing a beta-VAE is very straightforward, by just adding the beta weight in the loss equation and storing the parameter for traceability, so this did not take a lot of time.
+
+What is coming up next week
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next week I wanted to tinker around a bit with this parameter to see how it affects the quality of the reconstructions and the organization of the latent space, but I don't think this is an effective strategy, nor the priority. Thus, I will
+start implementing the conditional VAE, which will allow me to generate new streamlines by conditioning the latent space with a specific continuous variable.
```
Comment (jhlegarreta) [inline posts/2024/2024_07_19_Inigo_week_8.rst:32]: Unnecessary line break

File: posts/2024/2024_07_26_Inigo_week_9.rst [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+However, there are many methods to condition a Variational AutoEncoder, and they usually depend on the type of variable we want to condition on, so the methods for categoric variables (cat vs. dog, bundle_1_fiber vs. bundle_2_fiber, etc.) and continuous ones (age of the person, length of a streamline) are normally not applicable to both types.
+In the case of the FiberCup dataset, I chose to condition the latent space on the length of the streamlines, which is a continuous variable and it is a fairly easy thing to learn from the morphology of the streamlines.
```
Comment (jhlegarreta) [inline posts/2024/2024_07_26_Inigo_week_9.rst:21]: If new paragraph, add a blank line between them.

**Review by jhlegarreta — COMMENTED** [review]
Some minor style aspects pointed inline.

Thanks for the blog posts.

**Comment by skoudoro** [comment]
> I investigated it on the internet but I cannot find a reasonable solution.

No idea, maybe `numpydoc` is not correctly installed

> On the other hand, the CI fails but I am not familiar with this error:

interesting error... I will investigate later today. Sorry, it has a lower priority in my todolist. Maybe @deka27 can look at it?  check pydata_sphinx_theme requirement, and if we use a newer version. maybe we need to pin the version

File: posts/2024/2024_07_26_Inigo_week_9.rst [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+As a refresher, the main idea behind conditional generative models is being able to generate new data with some desired properties or parameters. To achieve this, it is common to organize the latent representation in a way that allows locating the regions in which the desired properties are found.
+
+For example, imagine our VAE learned a latent representation of images of cats with different fur lengths. If we do not condition our latent space on the fur length, our model might not learn about this distinctive feature found in the data space, and cats with drastically different fur lengths may be closely clustered together in the latent space. But with conditioning, we can tell the model to cluster the images along a "fur-length" dimension, so if we sample 2 images from a line that varies along that dimension but in opposite sides, we get a cat with very short fur, and another one, with very long fur. This results in a generative process that can be tuned on demand!
```
Comment (robinroy03) [inline posts/2024/2024_07_26_Inigo_week_9.rst:18]: This is a fun example :) I should also include more examples like this on my blogs haha (my work is also similar, but it's embedding text on latent space)

File: posts/2024/2024_07_26_Inigo_week_9.rst [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+After implementing the conditional VAE as in the provided reference and training it for 64 epochs (early stopped due to lack of improvement in the MSE) I got a very crappy looking reconstruction, but the latent space seems to be organized differently compared to the vanilla VAE, which suggests that the conditioning is doing something (good or not, we will see...).
+
+.. image:: /_static/images/gsoc/2024/inigo/cvae_first_reconstruction_result.png
+    :alt: First reconstruction of the training data of the conditional VAE (cVAE).
+    :width: 600
+
+On the other hand, note that the FiberCup has 7 distinct bundles, and both latent spaces show (when 2D-projected with the t-SNE algorithm) 7 clusters, suggesting that the network does know about the different bundles. Samples/streamlines are colored according to their length, and even if the bundle to which they belong is not plotted, we know that each cluster is formed by streamlines of different bundles because each bundle has a distinctive length.
```
Comment (robinroy03) [inline posts/2024/2024_07_26_Inigo_week_9.rst:29]: I have been thinking about this lately, FURY will be a great fit for t-SNE viz. Should make a demo/function. Not sure yet.

**Review by robinroy03 — APPROVED** [review]
Good progress :)
I've not rendered the blog, so can't comment on that. But the content is good, interesting work.

File: posts/2024/2024_07_26_Inigo_week_9.rst [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    - Checking that the predicted conditioning variable matches the input. In other words, measuring the MSE between the "true age" VS the "predicted age", their correlation, and plotting one against the other. This way you know how good the encoder is capturing the conditioning variable variation in the training data. The authors of the paper I based my implementation on do it like this:
+
+    .. image:: /_static/images/gsoc/2024/inigo/conditioning_validation_using_mse.png
+        :alt: Scatter plot of true vs predicted conditioning variable for validation.
+        :width: 600
+
+    - Visual checking fiber generation for specific bundles. Knowing that different bundles have different fiber lengths, we try to generate fibers of specific length, and see whether the generated fibers belong to the desired bundle (no matter if they are plausible or implausible).
+    Having length as the conditioning variable allows us to perform this trick, what would not be so intuitive to check if we had used Fractional Anisotropy or other DTI-derived metrics, as these are not visually as intuitive as length.
```
[the remark this answers is not in the document]
Comment (itellaetxe) [reply posts/2024/2024_07_26_Inigo_week_9.rst:51]: It is rendered as a full stop, no line breaks. I put it there for making text edition easier

File: posts/2024/2024_07_26_Inigo_week_9.rst (hunk 6 above)
Comment (itellaetxe) [reply posts/2024/2024_07_26_Inigo_week_9.rst:18]: thanks!

File: posts/2024/2024_07_26_Inigo_week_9.rst (hunk 7 above)
Comment (itellaetxe) [reply posts/2024/2024_07_26_Inigo_week_9.rst:29]: What are your ideas on that? Like implementing an interactive visualization for t-SNE?

Comment (robinroy03) [reply posts/2024/2024_07_26_Inigo_week_9.rst:30]: Yes

**Review by pjsjongsung — APPROVED** [review]

**Pull request merged**